extensions/frontend-tiles/src/BatchedTile.ts

```diff
@@ -137,8 +140,9 @@ export class BatchedTile extends Tile {
   public override async requestContent(_isCanceled: () => boolean): Promise<TileRequest.Response> {
     const url = new URL(this.contentId, this.batchedTree.reader.baseUrl);
     url.search = this.batchedTree.reader.baseUrl.search;
-    const response = await fetch(url.toString());
-    return response.arrayBuffer();
+    const response = await this.channel.fetch(url.toString(), fetch);
```

Comment:
Here is the usage of the new fetch function. For now, I kept this as a part of the tile request channel. No matter what, to access the db like this, we need these tiles to have a member with access to the db. Since the channel already existed and has access to the db, I kept this functionality as a part of the channel. It might be better to revert the channel being used by the batched tiles to a standard TileRequestChannel, and instead give the tiles a member that is an instance of IDBStorage, or some other connector class that communicates between the two. This shouldn't be a difficult change, but needs further discussion.

Reply:
As we discussed, I'd keep this simple, no need to add additional channels here, there is already a channel.
I'd remove the new channel. 
Branch this branch and let's try it out without channels, you will see the code will be more simple and elegant IMO.